### ISSUE TITLE
[java] GuardLogStatement: False positive with compile-time constant arguments

### ISSUE THREAD

**Issue by Myllyenko** [pmd/pmd#957]
**Affects PMD Version:** 6.1.0 (and other ones I think)

**Rule:** [GuardLogStatement](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#guardlogstatement)

**Description:** As stated in the rule's description, the rule is intended for helping programmers to 'skip the associate String creation and manipulation' but currently it triggers at any `String + String`-like construct even if the resulting value is actually a literal and no `Object` creation is involved at all.

**Code Sample demonstrating the issue:**

First, the sample from the rule's description:
```java
log.debug("log something" + " and " + "concat strings"); // the same as "log something and concat strings"
```

Second, more advanced case:
```java
final String constantPrefix = "log something";
final String constantSuffix = "concat strings";
log.debug(constantPrefix + " and " + constantSuffix ); // the same as "log something and concat strings"
```

**Running PMD through:** Doesn't matter

**Comment by jsotuyod** [comment]
@Myllyenko an artifact of working with code and not bytecode... but we can definitely work these cases out.

**Comment by Myllyenko** [comment]
Any chances that someone will look into this issue? :)
